plugins/browser-plugin-snowplow-ecommerce/src/ga4/api.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  trackProductView,
+  trackProductListView,
+  trackProductListClick,
+  trackCheckoutStep,
+  trackTransaction,
+  trackAddToCart,
+  trackRemoveFromCart,
+  trackPromotionClick,
+  trackPromotionView,
+} from '../api';
+import { Currency, GA4EcommerceObject, ItemList, Transaction, Promotion } from './types';
+import { transformG4ItemsToSPProducts, transformGA4PromotionToSPPromotion } from './utils';
+
+type Options = {
+  currency?: string;
+};
+
+const DEFAULT_PAYMENT_METHOD = 'unknown';
+
+export function trackGA4ViewCart() {
+  console.log('This function could be similarly tracked with the `setPageType` or `trackCheckoutStep` API.');
+}
+
+export function trackGA4ViewPromotion(ecommerce: GA4EcommerceObject & Promotion) {
+  const promotion = transformGA4PromotionToSPPromotion(ecommerce);
+  trackPromotionView(promotion);
+}
+
+export function trackGA4SelectPromotion(ecommerce: GA4EcommerceObject & Promotion) {
+  const promotion = transformGA4PromotionToSPPromotion(ecommerce);
+  trackPromotionClick(promotion);
+}
+
+export function trackGA4AddToCart(
+  ecommerce: GA4EcommerceObject & Currency,
+  opts: Options & { finalCartValue: number }
+) {
+  const currency = (ecommerce.currency || opts.currency)!;
```

Comment:
Same as for UA, I think now that we have `opts` it's easy enough to specify a currency if you don't have it in `ecommerce` already, so having an explicit fallback default should be OK now and we don't need to risk undefined currency with `!`.

Reply:
Same comment as with GA.